src/cdk/table/table.spec.ts

```diff
@@ -1687,6 +1707,55 @@ class NullDataCdkTableApp {
 class MultipleHeaderFooterRowsCdkTableApp {
 }
 
+@Component({
+  template: `
+    <cdk-table [dataSource]="dataSource" [trackBy]="trackByIndex">
+      <ng-container cdkColumnDef="column_a">
+        <cdk-header-cell *cdkHeaderCellDef> Column A</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row"> {{row.a}}</cdk-cell>
+      </ng-container>
+
+      <ng-container cdkColumnDef="column_b">
+        <cdk-header-cell *cdkHeaderCellDef> Column B</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row"> {{row.b}}</cdk-cell>
+      </ng-container>
+
+      <ng-container cdkColumnDef="column_c">
+        <cdk-header-cell *cdkHeaderCellDef> Column C</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row"> {{row.c}}</cdk-cell>
+      </ng-container>
+
+      <cdk-header-row *cdkHeaderRowDef="columnsToRender"></cdk-header-row>
+      <cdk-row *cdkRowDef="let row; columns: columnsToRender; when: when1" class="row-version-1">
+      </cdk-row>
+      <cdk-row *cdkRowDef="let row; columns: columnsToRender; when: when2" class="row-version-2">
+
+      </cdk-row>
+    </cdk-table>
+  `
+})
+class WhenRowChangeDetectionCdkTableApp {
+  dataSource: FakeDataSource = new FakeDataSource();
+  columnsToRender = ['column_a', 'column_b', 'column_c'];
+
+  showRow1 = true;
+
+  trackByIndex = (index: number, item: TestData) => index;
+  when1 = (index: number, _rowData: TestData) => this.showRow1;
+  when2 = (index: number, _rowData: TestData) => !this.showRow1;
```

Comment:
Instead of names like `when1` and `when2`, could you use something more meaningful? I find that tests are easier to read when they use real concepts, event if those concepts are silly (e.g. something like `isChocolate`, `isVanilla`)